staging/src/k8s.io/apiserver/pkg/endpoints/filters/disable_compression.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// ShouldDisableCompressionChecker checks if a given request should have compression disabled.
+type ShouldDisableCompressionChecker interface {
+	Check(*http.Request) (bool, error)
```

Comment:
check is a strange term here. This is not checking. It is a predicate.

Why is this an interface and not just a simple func?
```
type DisableCompressionFunc func(req) (bool, err)
```